### ISSUE TITLE
App crashes when category that is referred in linked categories no longer exists.

### ISSUE THREAD

**Issue by alexeykomov** [alexeykomov/green-travel#18]
Steps: 
1. Have items in history.
2. One of the search items in Core Data should have linked categories: `item.details.categoryIdToItems`.
3. Category that is linked must be deleted from new API payload.
3. New items from API should contain the same item with empty `include` field.
4. Press this item from search history.

Expected:
1. Details page of item should open normally without linked categories.

Actual:
1. App crashes when attempting to insert nil category (`[weakSelf.categories addObject:category];`).
```
self.categoryIdToItems = categoryIdToItems;
[self.categories removeAllObjects];
    
NSMutableDictionary<NSString *, Category *> *categoryUUIDToCategoryMap =  flattenCategoriesTreeIntoCategoriesMap(self.indexModel.categories);
[categoryIdToItems enumerateObjectsUsingBlock:^(CategoryUUIDToRelatedItemUUIDs * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
    Category *category = categoryUUIDToCategoryMap[obj.categoryUUID];
    [weakSelf.categories addObject:category];
}];
```

**Comment by alexeykomov** [comment]
Possible fix - do not store whole item in `correspondingPlaceItem` in `StoredSearchItem`, but store only uuid of referred item.

**Comment by alexeykomov** [comment]
Closed in 3955867.
